src/apps/SafeConnect/hooks/useWalletConnect.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState, useCallback, useEffect } from "react";
+import WalletConnect from "@walletconnect/client";
+import { IClientMeta } from "@walletconnect/types";
+import { useSafe } from "@rmeissner/safe-apps-react-sdk";
+
+export const LOCAL_STORAGE_URI_KEY = "safeAppWcUri";
+
+const useWalletConnect = () => {
+  const safe = useSafe();
+  const [wcClientData, setWcClientData] = useState<IClientMeta | null>(null);
+  const [connector, setConnector] = useState<WalletConnect | undefined>();
+
+  const wcDisconnect = useCallback(async () => {
+    connector?.killSession();
+    localStorage.removeItem(LOCAL_STORAGE_URI_KEY);
+    setConnector(undefined);
+    setWcClientData(null);
+  }, [connector]);
+
+  const wcConnect = useCallback(
+    async (uri: string) => {
+      const connector = new WalletConnect({
+        uri,
+        clientMeta: {
+          description: "Gnosis Safe",
+          url: "https://gnosis-safe.io/app",
+          icons: ["https://walletconnect.org/walletconnect-logo.png"],
+          name: "Gnosis Safe",
+        },
+      });
+      setConnector(connector);
+      setWcClientData(connector.peerMeta);
+      localStorage.setItem(LOCAL_STORAGE_URI_KEY, uri);
+
+      connector.on("session_request", (error, payload) => {
+        if (error) {
+          throw error;
+        }
+        debugger;
+        connector.approveSession({
+          accounts: [safe.getSafeInfo().safeAddress],
+          chainId: safe.getSafeInfo().network === "rinkeby" ? 4 : 1,
```

Comment:
This logic is something we need to overthing everywhere as we want to start support for "xdai" ... also in this app we do not limit it to `rinkeby` and `mainnet` (as it was adjusted for compound for exmaple)

Reply:
Ok, I'll split it up. 
Moving to a Monorepo will take some time so it should be handled in a different issue.